### ISSUE TITLE
Inventor Infusions Disappearing (Plus Same Controller Issue)

### ISSUE THREAD

**Issue by JandA-Forever** [SolastaMods/SolastaUnfinishedBusiness#2777]
Hello,

I am playing an online, 2-player multiplayer 6-Charactor Campaign, with one Bolgrif Inventor Grenader (Multi-Class is on, but I have only put levels into the Inventor). All UB mod classes, subclasses, feats, etc. are active. Characters just got to level 5. 

For the Inventor, when first selecting Infusions at level 2, there were 4 to select as described. However, with each subsequent level, an infusion selection disappears. I just finished his level 5 selections, and there were only 2 infusion selection available, and now that I am back on the normal game screen, when selecting Infusions from the Character Action bar, there is only 1 infusion remaining to be selected. I have terminated infusions a couple of times after long rests, so maybe I am misunderstanding what is supposed to be happening with Infusion. If that is the case, I do apologize for my ignorance.

![image](https://user-images.githubusercontent.com/120819238/226230461-8809e8dd-08bf-4134-9442-9c5db6f447dc.png)

Also, we are experiencing similar controller issues with our spell casters during battles. When we try to highlight the spell action, the display seems to desync and doesn't highlight Cast Spell. I will add another response if I can get a save where this issue is actively occurring. Keyboard and Mouse does work fine for everything, but we typically play with controllers/gamepads.

[Inventor Broke + Ctrlr Issue.zip](https://github.com/SolastaMods/SolastaUnfinishedBusiness/files/11013110/Inventor.Broke.%2B.Ctrlr.Issue.zip)

Thank you for this amazing mod!

**Comment by ThyWoof** [comment]
Have you tried RESPEC this hero? I cannot reproduce on my end.

**Comment by JandA-Forever** [comment]
I went through the RESPEC and it does seem to fix the issue for now. Since I originally experienced infusions disappearing each level, I tried RESPEC and then leveling up once, activating infusions, terminating infusions, then leveling up again, and I could not recreate the problem. All four infusions were still available after each level (up to level 5).

If I run into this again during normal play and find a consistent way to recreate, I will let you know. For now, RESPEC is the solution. Thank you!
